pkg/p2p/libp2p/stream.go

```diff
@@ -29,5 +32,28 @@ func (s *stream) Headers() p2p.Headers {
 }
 
 func (s *stream) FullClose() error {
-	return helpers.FullClose(s)
+	if err := s.CloseWrite(); err != nil {
+		_ = s.Reset()
+		return err
+	}
+	// So we don't wait forever
+	_ = s.SetDeadline(time.Now().Add(5 * time.Second))
```

Comment:
This is too short timeout. I would suggest to have it 60s as it was in earlier versions of helper package, maybe to lower it to 30s, as this deadline is the worst case scenario, and 5s is pretty short.